### PR TITLE
[LLM INFER] Fix some bugs and chatglm_v2 support block_attn

### PR DESCRIPTION
**Pull request by yuanlehome**
### PR types

New features
### PR changes

Models
### Description

- chatglm_v2 support block_attn mode, but the accuracy needs to be aligned
- 修复先前disable掉的诸多单测
- 略微优化下组网代码
- 添加USE_FASTER_TOP_P_SAMPLING环境变量用于使用性能更好的top_p_sampling算子

### CODE REVIEW

**Review by qingqing01 — APPROVED** [review]

File: llm/predict/predictor.py [hunk 1]
```diff
@@ -134,16 +134,16 @@ class PredictorArgument:
...
     )
 
-    @property
-    def total_max_length(self):
-        if self.device == "npu":
-            return self.src_length + self.max_length
-        else:
-            return 8192  # Maximum sequence length.
+    total_max_length: int = field(
+        default=4096, metadata={"help": "Super parameter. Maximum sequence length(encoder+decoder)."}
```
Comment (ZHUI) [inline llm/predict/predictor.py:138]: 这个跟npu相关同学确认的吗?

File: llm/predict/predictor.py [hunk 2]
```diff
@@ -520,7 +520,7 @@ def _preprocess(self, source):
             alibi_slopes = llm_utils.get_alibi_slopes(self.model_config.n_head)
             inputs["position_ids"] = paddle.to_tensor(alibi_slopes, dtype="float32")
             arange_tensor_encoder = paddle.arange(self.config.total_max_length, dtype=self.config.dtype)
-            alibi = alibi_slopes[None, :, None, None] * arange_tensor_encoder
+            alibi = (alibi_slopes[None, :, None, None] * arange_tensor_encoder).astype(self.config.dtype)
```
Comment (ZHUI) [inline llm/predict/predictor.py:524]: emm,这个 config dtype保险吗？用户可以改这个值。要不用里面一个tensor的dtype。

File: llm/predict/predictor.py [hunk 3]
```diff
@@ -1352,13 +1352,19 @@ def create_predictor(
...
                 model.eval()
-
             elif "chatglmv2forcausallm" in config.architectures[0].lower():
-                from paddlenlp.experimental.transformers import (
-                    ChatGLMv2ForCausalLMInferenceModel as Model,
-                )
-
-                model = Model.from_pretrained(
+                predictor_args.total_max_length = config.seq_length
+                if predictor_args.block_attn:
```
Comment (ZHUI) [inline llm/predict/predictor.py:1357]: emm，我建议吧 block_attn 放到config的属性里面，然后 ChatGLMv2InferenceModel 里面自己控制。
这里改的话，后期这样修改的模型太多了。

File: llm/predict/predictor.py (hunk 2 above)
Comment (yuanlehome) [reply llm/predict/predictor.py:524]: 这个dtype确实需要与config.dtype保持一致的

File: llm/predict/predictor.py (hunk 1 above)
Comment (yuanlehome) [reply llm/predict/predictor.py:138]: 已确认，没问题。

File: llm/predict/predictor.py (hunk 3 above)
Comment (yuanlehome) [reply llm/predict/predictor.py:1357]: 但严格来说其实这个不属于每个模型的Config，如果加入如LlamaConfig的话，每个模型的Config里都需要加，先保持这样吧，后面重构的时候，会看下有没有更好的方式
